string pack
---
string[] node_list
